src/VisualStudio/TestUtilities/IntegrationHelper.cs

```diff
@@ -0,0 +1,241 @@
+        {
+            using (var webClient = new WebClient())
+            {
+                await webClient.DownloadFileTaskAsync(downloadUrl, fileName).ConfigureAwait(continueOnCapturedContext: false);
+            }
+        }
+
+        public static IntPtr GetModalWindowFromParentWindow(IntPtr parentWindow)
+        {
+            foreach (var topLevelWindow in GetTopLevelWindows())
+            {
+                if ((User32.GetParent(topLevelWindow) == parentWindow) ||
+                    (User32.GetWindow(topLevelWindow, User32.GW_OWNER) == parentWindow) ||
+                    (User32.GetAncestor(topLevelWindow, User32.GA_PARENT) == parentWindow))
+                {
+                    return topLevelWindow;
+                }
+            }
+
+            return IntPtr.Zero;
+        }
+
+        public static string GetTitleForWindow(IntPtr window)
+        {
+            var titleLength = User32.SendMessage(window, User32.WM_GETTEXTLENGTH, IntPtr.Zero, IntPtr.Zero);
+
+            if (titleLength == IntPtr.Zero)
+            {
+                return string.Empty;
+            }
+
+            var title = new StringBuilder(titleLength.ToInt32() + 1);
+
+            User32.SendMessage(window, User32.WM_GETTEXT, (IntPtr)(title.Capacity), title);
+            return title.ToString();
+        }
+
+        public static IEnumerable<IntPtr> GetTopLevelWindows()
+        {
+            var topLevelWindows = new List<IntPtr>();
+
+            var enumFunc = new User32.WNDENUMPROC((hWnd, lParam) => {
+                topLevelWindows.Add(hWnd);
+                return true;
+            });
+            User32.EnumWindows(enumFunc, IntPtr.Zero);
+
+            return topLevelWindows;
+        }
+
+        public static void KillProcess(Process process)
+        {
+            if ((process != null) && (!process.HasExited))
+            {
+                process.Kill();
+            }
+        }
+
+        public static void KillProcess(string processName)
+        {
+            foreach (var process in Process.GetProcessesByName(processName))
+            {
+                KillProcess(processName);
+            }
+        }
+
+        public static DTE LocateDteForProcess(Process process)
+        {
+            object dte = null;
+            IRunningObjectTable runningObjectTable = null;
+            IEnumMoniker enumMoniker = null;
+            IBindCtx bindContext = null;
+            var monikers = new IMoniker[1];
+            var vsProgId = IntegrationHost.VsProgId;
+
+            Ole32.GetRunningObjectTable(0, out runningObjectTable);
+            runningObjectTable.EnumRunning(out enumMoniker);
+            Ole32.CreateBindCtx(0, out bindContext);
+
+            do
+            {
+                monikers[0] = null;
+
+                var monikersFetched = 0u;
+                var result = enumMoniker.Next(1, monikers, out monikersFetched);
+
+                if (result != 0)
```

Comment:
We should probably distinguish S_OK, S_FALSE, and an actual error here. In the error case, let's throw since that's catastrophic.